contracts/finance/andromeda-set-amount-splitter/src/contract.rs

```diff
@@ -0,0 +1,315 @@
+            ensure!(
+                lock_time.get_time(&env.block).seconds() <= ONE_YEAR,
+                ContractError::LockTimeTooLong {}
+            );
+            Splitter {
+                recipients: msg.recipients.clone(),
+                lock: lock_time.get_time(&env.block),
+            }
+        }
+        None => {
+            Splitter {
+                recipients: msg.recipients.clone(),
+                // If locking isn't desired upon instantiation, it's automatically set to 0
+                lock: Milliseconds::default(),
+            }
+        }
+    };
+    // Save kernel address after validating it
+
+    SPLITTER.save(deps.storage, &splitter)?;
+
+    let inst_resp = ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        &deps.querier,
+        info,
+        BaseInstantiateMsg {
+            ado_type: CONTRACT_NAME.to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            kernel_address: msg.kernel_address.clone(),
+            owner: msg.owner.clone(),
+        },
+    )?;
+
+    msg.validate(deps.as_ref())?;
+
+    Ok(inst_resp)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    if msg.result.is_err() {
+        return Err(ContractError::Std(StdError::generic_err(
+            msg.result.unwrap_err(),
+        )));
+    }
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ctx = ExecuteContext::new(deps, info, env);
+
+    match msg {
+        ExecuteMsg::AMPReceive(pkt) => {
+            ADOContract::default().execute_amp_receive(ctx, pkt, handle_execute)
+        }
+        _ => handle_execute(ctx, msg),
+    }
+}
+
+pub fn handle_execute(mut ctx: ExecuteContext, msg: ExecuteMsg) -> Result<Response, ContractError> {
+    let action_response = call_action(
+        &mut ctx.deps,
+        &ctx.info,
+        &ctx.env,
+        &ctx.amp_ctx,
+        msg.as_ref(),
+    )?;
+    let res = match msg {
+        ExecuteMsg::UpdateRecipients { recipients } => execute_update_recipients(ctx, recipients),
+        ExecuteMsg::UpdateLock { lock_time } => execute_update_lock(ctx, lock_time),
+        ExecuteMsg::Send {} => execute_send(ctx),
+        _ => ADOContract::default().execute(ctx, msg),
+    }?;
+    Ok(res
+        .add_submessages(action_response.messages)
+        .add_attributes(action_response.attributes)
+        .add_events(action_response.events))
+}
+
+fn execute_send(ctx: ExecuteContext) -> Result<Response, ContractError> {
+    let ExecuteContext { deps, info, .. } = ctx;
+
+    // Maximum of two coins allowed for now
+    ensure!(
+        info.funds.len().le(&2),
+        ContractError::ExceedsMaxAllowedCoins {}
+    );
+
```

Comment:
I think some validation on the amount sent would be beneficial here so that an error can be thrown to the user. They should only ever send as much as is required to pay all the recipients exactly so that no amount is left in the contract.